aws-datastore/src/test/java/com/amplifyframework/datastore/appsync/AppSyncMocking.java

```diff
@@ -213,91 +224,159 @@ public static OnDeleteConfigurator onDelete(AppSync mock) {
+         * @param nextToken nextToken that should be used to build the requestForNextResult on the GraphQLResponse.
+         * @param responseItems The items that should be included in the mocked response, for the model class
+         * @param <M> Type of models for which a response is mocked
+         * @return The same Configurator instance, to enable chaining of calls
+         * @throws AmplifyException if a ModelSchema cannot be created in order to build the sync request.
+         */
+        @SuppressWarnings("varargs")
+        @SafeVarargs
+        public final <M extends Model> SyncConfigurator mockSuccessResponse(
+                Class<M> modelClass,
+                String token,
+                String nextToken,
+                ModelWithMetadata<M>... responseItems) throws AmplifyException {
+            final Iterable<ModelWithMetadata<M>> items = new HashSet<>(Arrays.asList(responseItems));
+            AppSyncGraphQLRequest<PaginatedResult<ModelWithMetadata<M>>> requestForNextResult = null;
+            if (nextToken != null) {
+                requestForNextResult = AppSyncRequestFactory.buildSyncRequest(modelClass, null, null)
+                        .newBuilder()
+                        .variable("nextToken", "String", nextToken)
+                        .build();
+            }
+            return mockSuccessResponse(
+                    matcherFor(modelClass, token),
+                    new GraphQLResponse<>(
+                            new PaginatedResult<>(items, requestForNextResult),
+                            Collections.emptyList()
+                    )
+            );
+        }
+
+        /**
+         * Configures an instance of an {@link AppSync} to invoke the response callback with the provided mockResponse
+         * when asked to {@link AppSync#sync(GraphQLRequest, Consumer, Consumer)}.
+         * @param requestMatcher ArgumentMatcher which returns true if the GraphQLRequest should be mocked.
+         * @param mockResponse GraphQLResponse to be passed back in the response callback.
+         * @param <M> Type of models for which a response is mocked
+         * @return The same Configurator instance, to enable chaining of calls
+         */
+        public <M extends Model> SyncConfigurator mockSuccessResponse(
+                ArgumentMatcher<GraphQLRequest<PaginatedResult<ModelWithMetadata<M>>>> requestMatcher,
+                GraphQLResponse<PaginatedResult<ModelWithMetadata<M>>> mockResponse) {
             doAnswer(invocation -> {
                 // Get a handle to the response consumer that is passed into the sync() method
-                // Response consumer is the third param, at index 2 (@0, @1, @2, @3).
-                final int argumentPositionForResponseConsumer = 2;
-                final Consumer<GraphQLResponse<Iterable<ModelWithMetadata<T>>>> consumer =
+                // Response consumer is the second param, at index 1 (@0, @1, @2).
```

Comment:
I really appreciate you taking the time and consideration to update some of this ancillary stuff. This would be easy to overlook, and often is.